api/apiroutes.go

```diff
@@ -1734,6 +1741,41 @@ func (c *appContext) addressTotals(w http.ResponseWriter, r *http.Request) {
 	writeJSON(w, totals, indent)
 }
 
+// addressExists provides access to the existsaddresses RPC call and parses the
+// hexadecimal string into a list of bools. A maximum of 64 addresses can be
+// provided. Duplicates are not filtered.
+func (c *appContext) addressExists(w http.ResponseWriter, r *http.Request) {
+	indent, err := c.getIndentQuery(r)
+	if err != nil {
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	addresses, err := m.GetAddressRawCtx(r, c.Params, maxExistAddrs)
+	if err != nil {
+		apiLog.Errorf("addressExists rejecting request: %v", err)
+		http.Error(w, "address parsing error", http.StatusBadRequest)
+		return
+	}
+	// GetAddressCtx throws an error if there would be no addresses.
+	strMask, err := c.nodeClient.ExistsAddresses(addresses)
+	if err != nil {
+		log.Warnf("existsaddress error: %v", err)
+		http.Error(w, http.StatusText(422), 422)
+	}
+	b, err := hex.DecodeString(strMask)
+	if err != nil {
+		log.Warnf("existsaddress error: %v", err)
+		http.Error(w, http.StatusText(422), 422)
+	}
+	mask := binary.LittleEndian.Uint64(append(b, make([]byte, 8-len(b))...))
+	exists := make([]bool, 0, len(addresses))
+	for n := range addresses {
+		exists = append(exists, (mask&(1<<uint8(n))) != 0)
+	}
```

Comment:
Looks good.  My only concern is that `maxExistAddrs` cannot be more than 64 because of this, and there's no check on `maxExistAddrs`.  I worry about someone bumping up `maxExistAddrs` in the future without understanding it would lead to a panic in `make` here when >64 addresses get passed, especially because it might appear to work OK in many cases.

Reply:
Should I just document `maxExistAddr` better?

Reply:
Yah, that's cool.  I think a note there not to go beyond 64 without modifying addressExists (or whatever) would be sufficient to prevent future breakage.